src/main/java/org/openrewrite/staticanalysis/EqualsAvoidsNullVisitor.java

```diff
@@ -90,21 +96,19 @@ private boolean hasCompatibleArgument(J.MethodInvocation m) {
     }
 
     private boolean isStringComparisonMethod(J.MethodInvocation methodInvocation) {
-        return EQUALS.matches(methodInvocation) ||
-               !style.getIgnoreEqualsIgnoreCase() &&
-               EQUALS_IGNORE_CASE.matches(methodInvocation) ||
-               COMPARE_TO.matches(methodInvocation) ||
-               COMPARE_TO_IGNORE_CASE.matches(methodInvocation) ||
-               CONTENT_EQUALS.matches(methodInvocation);
+        return METHOD_MATCHERS.stream().anyMatch(matcher -> matcher.matches(methodInvocation))
```

Comment:
While I understand the appeal of `anyMatch`, we try to limit our use of the streams API due to the new objects it creates when executing, which adds up at scale.

Reply:
resolved, thanks for the hint.